### ISSUE TITLE
d3 4.x support?

### ISSUE THREAD

**Issue by tristanguigue** [markmarkoh/datamaps#320]
This library does not seem to be compatible with the latest version of d3 for example the API of d3.geo has changed quite a lot: https://github.com/d3/d3/blob/master/API.md#geographies-d3-geo

Is there plans for d3 4.x support?

**Comment by markmarkoh** [comment]
@tristanguigue you're right, d3's namespace flattening won't work nicely with Datamaps.

I'll work on an upgrade (or dual support for both 3.x and 4.x).

**Comment by Antho2407** [comment]
Any news on D3 4.X support ? Would be awesome !

**Comment by cnsgithub** [comment]
I agree, having datamaps support D3 v4 would be great as we already use other v4 based D3 libs and are not able to include two different versions. Thanks for any news.

**Comment by wkurniawan07** [comment]
Hi @markmarkoh, any updates on this? Would be great if this can be rolled out soon!

**Comment by stickywes** [comment]
There's always the PR!

**Comment by markmarkoh** [comment]
Hey All! I'd love to support d3 v4. We'll have to rethink some of the way we package this library up since it's not a trivial update.

I've created a Github project to track work here. I'm happy to take any support if any of you have some d3 v4 experience and some extra cycles laying around.

https://github.com/markmarkoh/datamaps/projects/1

**Comment by ZhengLuke** [comment]
Any news on the updated version that supports d3 v4? Thanks so much!!

**Comment by aaani** [comment]
Any updates?

**Comment by davca92** [comment]
Hello!
Is there a chance to get it done? :)
Thanks! :)

**Comment by jawadahmadd** [comment]
Any news for d3 v4 support ?

**Comment by mojoaxel** [comment]
PLEASE STOP ASKING about a status update. Just subscript to this issue!

Supporting D3 version 4 [is quite a change](https://github.com/markmarkoh/datamaps/projects/1). Maybe we can fund @markmarkoh to work on it!?
https://www.bountysource.com/issues/35788460-d3-4-x-support

**Comment by Move2Create** [comment]
Also, there is an easy workaround, https://stackoverflow.com/questions/16156445/multiple-versions-of-a-script-on-the-same-page-d3-js#16156569

It has worked for me so far, but I'll leave it to the more qualified to determine if this is a good solution.

**Comment by kelvinau** [comment]
how about support for d3 v5?

**Comment by treyseals** [comment]
Still no v5 support? Has the project died?

**Comment by velum** [comment]
And now, d3 v4 support.

**Comment by traceypooh** [comment]
I'll link an example using full ESM `import`, latest: `d3`, `jquery`, and `topojson`, and such so you can see it in action at my work site once i get this pushed live

**Comment by bkenwright** [comment]
Just extending what traceypooh did, I wrote a small 'patch' script - which lets you upgrade the current datamaps.js.  

It essentially loads the script (as text), swap legacy names (e.g., mouse, d3.geo, ..) with the current v7 ones, then launches the script.

Working example
https://notebook.xbdev.net/index.php?page=d3datamap&

:)

Hopefully, they'll release a v7 update - but in the meantime it shouldn't be that hard to port over ^_^

Happy coding
